gatorgrade/generate/generate.py

```diff
@@ -1,21 +1,82 @@
-"""
+# Define colors for terminal output
+OKGREEN = "\033[92m"
+WARNING = "\033[93m"
+FAIL = "\033[91m"
 
-# Import the necessary libraries
 
-
-def create_targeted_paths_list():
+def create_targeted_paths_list(
+    key_word_list: List[str], relative_run_path: str = "."
+) -> List[str]:
     """Generate a list of targeted paths by walking the paths."""
+    targeted_paths = []
     # Go through the root repo, the sub dictionaries and files.
-    # Select only files in the dictionaries with specific names.
-    # Add those targeted file paths into a list and returns it.
+    # The os.walk will only scan the paths.
+    # So the empty folders containing nothing won't be gone through
+    for dirpath, _, filenames in os.walk(relative_run_path):
+        # Split path string into multiple layers of directories
+        path_dir_list = dirpath.split("/")
+        # Ignore folder starting with double underscore
+        if any(path.startswith("__") for path in path_dir_list):
+            continue
+        # Ignore hidden folders and first layer. the root repo is always dot
+        # Keep double dot. It represents the returning sign
+        if any(
+            path.startswith(".") and not path.startswith("..")
+            for path in path_dir_list[1:]
+        ):
+            continue
+        for filename in filenames:
+            # Ignore the file starting with double underscore and hidden file
+            if filename.startswith("__") or filename.startswith("."):
+                continue
+            # Add paths when they have the key words in the second and the third directories
+            # For the path with only two directories,
+            # check key words in the second directory folder name
+            if len(path_dir_list) == 2:
+                if path_dir_list[1] in key_word_list:
+                    targeted_paths.append(os.path.join(dirpath, filename))
+
+            # For the other paths with more than 2 directories,
+            # check key words in the second and third directories
+            elif len(path_dir_list) > 2:
+                if (
+                    path_dir_list[1] in key_word_list
+                    or path_dir_list[2] in key_word_list
+                ):
+                    targeted_paths.append(os.path.join(dirpath, filename))
+
+            if filename in key_word_list:
+                targeted_paths.append(os.path.join(dirpath, filename))
+
+    # If any of the user inputted file does not exist in any directory,
+    # throw an exception indicating failure
+    if not targeted_paths:
+        raise FileNotFoundError(
+            f"{FAIL}FAILURE: None of the user-provided file paths are"
+            + " found in the provided directory and the 'gatorgrade.yml' is NOT generated"
+        )
+
+    # If some of the files are found and some are not found,
+    # output a warning message saying which files were not found
+    targeted_paths_string = " ".join(targeted_paths)
+    for key in key_word_list:
```

Comment:
`key` and `key_word_list` don't help me much as the reader. What I'm gathering is that they are some way to "exlude" paths, is that correct?

Maybe we name these `excluded_path` and `excluded_path_list`, what do you think?

I absolutely could be misinterpreting the intent though!

Reply:
Yup, thanks for your advice. I thought about the same problem before as well and thought keyword was good enough. Here the keyword list is used to contain target and wanted ones. So I may change it to `included_path_list` or `targeted_path_list` Which one do you prefer. I may do the second one